### ISSUE TITLE
Cannot build the SDK using esbuild

### ISSUE THREAD

**Issue by unilynx** [aws/amazon-chime-sdk-js#2494]
### What happened and what did you expect to happen?

After upgrading from 2.15.0 (sorry) to 3.9.0 building the SDK using esbuild fails because it tries to import node-native modules (such as `buffer`) 

The issue looks very similar to https://github.com/aws/amazon-chime-sdk-js/issues/2455 (rollup failing 3.7.0)

### Have you reviewed our existing documentation?

- [X] GitHub [issues](https://github.com/aws/amazon-chime-sdk-js/issues)
- [X] FAQs [page](https://aws.github.io/amazon-chime-sdk-js/modules/faqs.html)
- [X] README [page](https://github.com/aws/amazon-chime-sdk-js#amazon-chime-sdk-for-javascript)
- [X] Feature-specific [guides](https://github.com/aws/amazon-chime-sdk-js#guides)

### Reproduction steps

```
npm i amazon-chime-sdk-js@^3.9.0
npm I esbuild
./node_modules/esbuild/bin/esbuild  --bundle amazon-chime-sdk-js
```

### Amazon Chime SDK for JavaScript version

3.9.0

### What browsers are you seeing the problem on?

esbuild

### Browser version

0.15.13

### Browser console logs

```
$ ./node_modules/esbuild/bin/esbuild  --bundle amazon-chime-sdk-js

✘ [ERROR] Could not resolve "buffer"

    node_modules/@aws-sdk/hash-node/dist-cjs/index.js:5:25:
      5 │ const buffer_1 = require("buffer");


(and more node builtin modules)
```

passing --format=esm to esbuild makes no difference

with --log-level=verbose I notice that esbuild uses the dist-cjs directory of client-chime-sdk-messaging but there's no browser override there to point it to runtimeConfig.js - that override exists only for the dist-es directory.

```
⬥ [VERBOSE] Resolving import "./runtimeConfig" in directory "/Users/arnold/experiment-aws-sdk/node_modules/@aws-sdk/client-chime-sdk-messaging/dist-cjs" of type "require-call"
…
  Primary path is "/Users/arnold/experiment-aws-sdk/node_modules/@aws-sdk/client-chime-sdk-messaging/dist-cjs/runtimeConfig.js" in namespace "file"
```

It has seen dist-es but is apparently not using it because of

```
  Resolved to "/Users/arnold/experiment-aws-sdk/node_modules/@aws-sdk/client-chime-sdk-messaging/dist-cjs/index.js" because of "require"
```

So I'm not sure who's at fault. Should client-chime-sdk-messaging offer a `module` version if it expects to be able to use the module version of aws-sdk/client-chime-sdk-messaging (which *does* apparently have the browser workaround ?

Or is aws-sdk/client-chime-sdk-messaging at fault for not providing a browser-compatible dist-cjs version ?

Or is esbuild at fault for not preferring the dist-es version?

**Comment by LeviSklaroff** [comment]
Hey sorry for the late response, I can't say with certainty on whether this is an issue with esbuild or the aws-sdk. For now are you able to use version 3.6.0 to see if that works fine for you? It's not ideal but should at least allow you to upgrade to a newer version. I've also went ahead and created a bug report for EsBuild and Aws-Sdk to see if a bug fix can be found https://github.com/evanw/esbuild/issues/2692 and https://github.com/aws/aws-sdk-js-v3/issues/4217

**Comment by unilynx** [comment]
Updating main-fields fixes 3.9.0 for me. I would have expected something else to break with such a change (as evanw indicates in https://github.com/evanw/esbuild/issues/2692#issuecomment-1322836181 ) but so far nothing I currently uses breaks with this change.
